pytorch_lightning/core/datamodule.py

```diff
@@ -373,10 +375,10 @@ def dataloader(ds, shuffle=False):
 
         def train_dataloader():
             if isinstance(train_dataset, Mapping):
-                return {key: dataloader(ds, shuffle=True) for key, ds in train_dataset.items()}
+                return {key: dataloader(ds, shuffle=shuffle) for key, ds in train_dataset.items()}
```

Comment:
Won't it be better to do: `shuffle=not isinstane(ds, IterableDataset)` directly ?
As you might have Dataset and IterableDataset in the list or dictionaries.

Reply:
That makes more sense. I'll push the change you mentioned.